src/bridge/trx/index.ts

```diff
@@ -126,25 +126,27 @@ export class TronBridge extends Bridge {
     );
   }
 
-  async approve(approveData: ApproveData): Promise<TransactionResponse> {
-    const { tokenAddress, spender, owner } = approveData;
+  async approve(params: ApproveParamsDto): Promise<TransactionResponse> {
+    const { tokenAddress, spender, owner, amount } = params;
     const tokenContract = await this.getContract(tokenAddress);
+    const amountHex = amount === undefined ? MAX_AMOUNT : amountToHex(amount);
     const transactionHash = await tokenContract
```

Comment:
Try to use buildRawTransactionApprove here

Reply:
I replaced custom send transaction logic with sending transaction with buildRawTransactionApprove method.